clients/include/testing_syevd_heevd.hpp

```diff
@@ -0,0 +1,779 @@
+                                 ltwork,
+                                 hE.data(),
+                                 sizeE,
+                                 iwork.data(),
+                                 liwork,
+                                 hinfo[b]);
+        *cpu_time_used = get_time_us_no_sync() - *cpu_time_used;
+    }
+
+    syevd_heevd_initData<true, false, T>(handle, evect, n, dA, lda, bc, hA, A, 0);
+
+    // cold calls
+    for(int iter = 0; iter < 2; iter++)
+    {
+        syevd_heevd_initData<false, true, T>(handle, evect, n, dA, lda, bc, hA, A, 0);
+
+        CHECK_ROCBLAS_ERROR(hipsolver_syevd_heevd(FORTRAN,
+                                                  handle,
+                                                  evect,
+                                                  uplo,
+                                                  n,
+                                                  dA.data(),
+                                                  lda,
+                                                  stA,
+                                                  dD.data(),
+                                                  stD,
+                                                  dWork.data(),
+                                                  lwork,
+                                                  dinfo.data(),
+                                                  bc));
+    }
+
+    // gpu-lapack performance
+    hipStream_t stream;
+    CHECK_ROCBLAS_ERROR(hipsolverGetStream(handle, &stream));
+    double start;
+
+    for(int iter = 0; iter < hot_calls; iter++)
+    {
+        syevd_heevd_initData<false, true, T>(handle, evect, n, dA, lda, bc, hA, A, 0);
+
+        start = get_time_us_sync(stream);
+        hipsolver_syevd_heevd(FORTRAN,
+                              handle,
+                              evect,
+                              uplo,
+                              n,
+                              dA.data(),
+                              lda,
+                              stA,
+                              dD.data(),
+                              stD,
+                              dWork.data(),
+                              lwork,
+                              dinfo.data(),
+                              bc);
+        *gpu_time_used += get_time_us_sync(stream) - start;
+    }
+    *gpu_time_used /= hot_calls;
+}
+
+template <bool FORTRAN, bool BATCHED, bool STRIDED, typename T>
+void testing_syevd_heevd(Arguments& argus)
+{
+    using S = decltype(std::real(T{}));
+
+    // get arguments
+    hipsolver_local_handle handle;
+    char                   evectC = argus.get<char>("jobz");
+    char                   uploC  = argus.get<char>("uplo");
+    int                    n      = argus.get<int>("n");
+    int                    lda    = argus.get<int>("lda", n);
+    int                    stA    = argus.get<int>("strideA", lda * n);
+    int                    stD    = argus.get<int>("strideW", n);
```

Comment:
In the API, I adopted cuSOLVER's naming of `W` instead of `D`. I figured that if we were to add a strided batch version, that the argument would come into hipsolver-bench as `strideW` instead of `strideD`.

I suppose I could just change it to use `D` as the name instead of `W`. It would cause less confusion for us.